### PR TITLE
refactor and abstract app.py

### PR DESCRIPTION
**Pull request by joshuacook**

### CODE REVIEW

**Comment by joshuacook** [comment]
@geekman2

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1 @@
 FROM jupyter/scipy-notebook
```
Comment (joshuacook) [inline Dockerfile:1]: this Dockerfile is officially irrelevant
we should just pull directly from `jupyter/scipy-notebook`

File: Makefile [hunk 2]
```diff
@@ -1,11 +1,18 @@
+wrangle_data:
+	docker build -t timing_comparison .
+	docker run -it -v $(shell pwd):/home/jovyan/work --rm timing_comparison python -m lib.data.wrangler
```
Comment (joshuacook) [inline Makefile:3]: `docker run -it -v $(shell pwd):/home/jovyan/work --rm jupyter/scipy-notebook python -m lib.data.wrangler`

File: Makefile [hunk 3]
```diff
@@ -1,11 +1,18 @@
+wrangle_data:
+	docker build -t timing_comparison .
```
Comment (joshuacook) [inline Makefile:2]: get rid of the build for each task

**Comment by dtimm** [comment]
Super giant structural changes. Merging.

**Pull request merged**